app/scripts/controllers/preferences.js

```diff
@@ -666,6 +667,24 @@ export default class PreferencesController {
     return Promise.resolve(domain);
   }
 
+  /**
+   * A setter for the `useLedgerLive` property
+   * @param {bool} domain - Value for ledger live support
+   * @returns {Promise<string>} A promise of the update to useLedgerLive
+   */
+  setLedgerLivePreference(useLedgerLive) {
+    this.store.updateState({ useLedgerLive });
+    return Promise.resolve(useLedgerLive);
```

Comment:
I wanted to follow the existing pattern, as seen by `setIpfsGateway`, `completeOnboarding`, etc.

Reply:
Ah I see. I think those were mistakes as well. Returning `Promise.resolve` from an otherwise synchronous function is bad.

If we want it to be async for compatibility reasons (e.g. maybe we want to leave the door open to do async things in here later or something), we should instead make the whole function `async` so that we have consistent error handling behaviour as well, rather than now where exceptions would get thrown synchronously rather than returned with a `Promise.reject`, which is _super_ unexpected/unintuitive for the caller.

But I don't see any case for this being async at all in this specific case. It seems inappropriate in the other examples as well.